### PR TITLE
More Ghostspawners

### PR DESCRIPTION
**Pull request by Geevies**
* Posibrains are now in the ghost spawner menu. Activating them is now a one-and-done deal, as they remain open in the ghost spawner until someone disables it or someone spawns in.
* Syndicate Cyborgs have been added to the ghost spawner menu, and will remain in a boot-up state until someone spawns into it.
* Syndicate Cyborg eye overlays have been fixed.
* Wizard Familiars have been added to the ghost spawner menu, same as above.
* Mining Drones have been added to the ghost spawner menu, one swipe is all you need, until someone joins in as them.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

**Comment by Geevies** [comment]
latest commit also fixes clientsize being set to a super huge number because of a null value, when you join into a wizard from the ghost spawner menu

File: code/game/gamemodes/antagspawner.dm [hunk 1]
```diff
@@ -22,22 +22,17 @@
...
-	spawn(600)
-		if(F)
-			if(!F.ckey || !F.client)
-				F.visible_message("With no working brain to keep \the [F] working, it is teleported back.")
-				qdel(F)
-				uses++
+	SSghostroles.add_spawn_atom("syndiborg", F)
+	var/area/A = get_area(src)
+	if(A)
+		say_dead_direct("A syndicate cyborg has started its boot process in [A.name]! Spawn in as it by using the ghost spawner menu in the ghost tab, and try to be good!")
```
Comment (NonQueueingMatt) [inline code/game/gamemodes/antagspawner.dm:38]: It automatically tells dead people when the ghost spawners gets activated, not really necessary?

File: code/modules/ghostroles/spawner/simplemob/wizard_familiar.dm [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	spawn_mob = /mob/living/simple_animal
+
+/datum/ghostspawner/simplemob/wizard_familiar/select_spawnpoint(var/use)
+	return TRUE //We just fake it here, since the spawnpoint is selected if someone is spawned in.
+
+//The proc to actually spawn in the user
+/datum/ghostspawner/simplemob/wizard_familiar/spawn_mob(mob/user)
+	if(!length(spawn_atoms))
+		to_chat(user, SPAN_DANGER("There are no available wizard familiars to spawn at!"))
+		return FALSE
```
Comment (NonQueueingMatt) [inline code/modules/ghostroles/spawner/simplemob/wizard_familiar.dm:17]: Nitpick but I don't like how this is copypasted all over

File: code/modules/mining/minebot.dm [hunk 3]
```diff
@@ -60,11 +61,22 @@
 /mob/living/silicon/robot/drone/mining/request_player()
 	if(too_many_active_drones())
 		return
-	var/datum/ghosttrap/G = get_ghost_trap("mining drone")
-	G.request_player(src, "Someone is attempting to reboot a mining drone.", 30 SECONDS)
+	seeking_player = TRUE
+	SSghostroles.add_spawn_atom("mining_drone", src)
+	var/area/A = get_area(src)
+	if(A)
+		say_dead_direct("Someone is attempting to reboot a mining drone in [A.name]! Spawn in as it by using the ghost spawner menu in the ghost tab.")
```
Comment (NonQueueingMatt) [inline code/modules/mining/minebot.dm:68]: Ditto above, not really necessary I think?

File: code/modules/mob/living/carbon/brain/posibrain.dm [hunk 4]
```diff
@@ -3,46 +3,51 @@
...
-
-/obj/item/device/mmi/digital/posibrain/attack_ghost(var/mob/abstract/observer/user)
-	if(!searching || (src.brainmob && src.brainmob.key))
+/obj/item/device/mmi/digital/posibrain/attack_self(mob/user)
+	if(brainmob.ckey)
+		to_chat(user, SPAN_WARNING("\The [src] already has an active occupant!"))
 		return
+	var/area/A = get_area(src)
+	if(brainmob && !brainmob.key && !searching)
+		if(!searching)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/carbon/brain/posibrain.dm:23]: You're already checking searching above, remove that from the if at the top

File: code/game/gamemodes/antagspawner.dm (hunk 1 above)
Comment (Geevies) [reply code/game/gamemodes/antagspawner.dm:38]: yeah, it doesn't tell them if subsequent mobs spawn, which is fairly useful IMO.

File: code/modules/ghostroles/spawner/simplemob/wizard_familiar.dm (hunk 2 above)
Comment (Geevies) [reply code/modules/ghostroles/spawner/simplemob/wizard_familiar.dm:17]: ehhhhhhhhhhh dunno if there's something i can do for that rn

File: code/modules/mob/living/carbon/brain/posibrain.dm [hunk 5]
```diff
@@ -3,46 +3,51 @@
...
-		M.show_message("<span class='notice'>The positronic brain buzzes quietly, and the golden lights fade away. Perhaps you could try again?</span>")
-
-/obj/item/device/mmi/digital/posibrain/attack_ghost(var/mob/abstract/observer/user)
-	if(!searching || (src.brainmob && src.brainmob.key))
+/obj/item/device/mmi/digital/posibrain/attack_self(mob/user)
+	if(brainmob.ckey)
+		to_chat(user, SPAN_WARNING("\The [src] already has an active occupant!"))
 		return
+	var/area/A = get_area(src)
+	if(brainmob && !brainmob.key && !searching)
```
Comment (Geevies) [inline code/modules/mob/living/carbon/brain/posibrain.dm:22]:
```suggestion
	if(brainmob && !brainmob.key)
```

**Pull request merged**